### PR TITLE
Update dma V1 binding with STM32 macros

### PR DESCRIPTION
**Pull request by djiatsaf-st**
- Add STM32 macro 15th bit for st,stm32-dma-v1.yaml , necessary for Peripheral Increment Offset Size configuration  
- Update  st,stm32-dma-v1.yaml  description . 

This make easy to configure dma peripherals.

### CODE REVIEW

File: dts/bindings/dma/st,stm32-dma-v1.yaml [hunk 1]
```diff
@@ -15,42 +15,46 @@ description: |
...
+               0x0: STM32_DMA_OFFSET_LINKED_BUS: offset size is linked to the peripheral bus width
+               0x1: STM32_DMA_OFFSET_FIXED_4: offset size is fixed to 4 (32-bit alignment)
         -bit 16-17 : Priority level
-               0x0: low
-               0x1: medium
-               0x2: high
-               0x3: very high
+               0x0: STM32_DMA_PRIORITY_LOW: low
+               0x1: STM32_DMA_PRIORITY_MEDIUM: medium
+               0x2: hSTM32_DMA_PRIORITY_HIGH: igh
```
Comment (mathieuchopstm) [inline dts/bindings/dma/st,stm32-dma-v1.yaml:46]: Typo:
```suggestion
               0x2: hSTM32_DMA_PRIORITY_HIGH: high
```

**Review by mathieuchopstm — DISMISSED** [review]
LGTM besides typo (see review comment).

Nit: maybe these comments should be moved to `stm32_dma.h`?
* reduces duplication among binding files
  * `st,stm32-dma-v1.yaml`
  * `st,stm32-dma-v2.yaml`
  * `st,stm32-dma-v2.bis.yaml`
  * `st,stm32-dma-u5.yaml`
* IDEs can display documentation while using the macros

cc @erwango what do you think?

File: dts/bindings/dma/st,stm32-dma-v1.yaml (hunk 1 above)
Comment (erwango) [reply dts/bindings/dma/st,stm32-dma-v1.yaml:46]: 2 typos actually:
```suggestion
               0x2: STM32_DMA_PRIORITY_HIGH: high
```

**Review by erwango — CHANGES_REQUESTED** [review]

**Comment by erwango** [comment]
> Nit: maybe these comments should be moved to stm32_dma.h?

Well, bindings are where all these are supposed to be documented https://docs.zephyrproject.org/latest/build/dts/api/bindings/dma/st%2Cstm32-dma-v1.html#dtbinding-st-stm32-dma-v1

**Review by erwango — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Comment by erwango** [comment]
@teburd Up to you (or you can assign it to me)

**Pull request merged**